harness/determined/_info.py

```diff
@@ -357,8 +361,12 @@ def container_rank(self) -> int:
         When using a distributed training framework, the framework may choose a different rank for
         this container.
         """
-        assert self._rendezvous_info is not None
-        return self._rendezvous_info.container_rank
+        # Presently, only trials are allowed to use the rendezvous API.
+        if self.task_type == "TRIAL":
```

Comment:
maybe invert this flow (`if self.task_type != "TRIAL"`) to match the surrounding code.